### PR TITLE
ambiguous naming fix

### PR DESCRIPTION
**Pull request by raulsi**
I updated he ambiguous naming of variables. check it out.  if there is a problem tell me i am also new to rust  :) - This will fix issue #3

### CODE REVIEW

**Comment by AtheMathmo** [comment]
Awesome! Thanks for taking the time to do this. I'll take a look now and make comments if I spot anything.

File: rusty-machine/tests/learning/gp.rs [hunk 1]
```diff
@@ -9,11 +9,11 @@ fn test_default_gp() {
 	gp.noise = 10f64;
 
 	let train_data = Matrix::new(10,1,vec![0.,1.,2.,3.,4.,5.,6.,7.,8.,9.]);
-	let target = Vector::new(vec![0.,1.,2.,3.,4.,4.,3.,2.,1.,0.]);
+	let targets = Vector::new(vec![0.,1.,2.,3.,4.,4.,3.,2.,1.,0.]);
 
 	gp.train(&train_data, &target);
```
Comment (AtheMathmo) [inline rusty-machine/tests/learning/gp.rs]: This line should have `targets` too.

File: rusty-machine/src/learning/lin_reg.rs [hunk 2]
```diff
@@ -35,20 +35,20 @@ impl SupModel<Matrix<f64>, Vector<f64>> for LinRegressor {
     /// let data = Matrix::new(3,2, vec![1.0, 2.0, 1.0, 3.0, 1.0, 4.0]);
```
Comment (AtheMathmo) [inline rusty-machine/src/learning/lin_reg.rs]: You need to update data and values to be inputs and targets too.

File: rusty-machine/src/lib.rs [hunk 3]
```diff
@@ -105,25 +105,25 @@ pub mod learning {
...
 
-        /// Predict output from data.
-        fn predict(&self, data: &T) -> U;
+        /// Predict output from inputs.
+        fn predict(&self, inputs: &T) -> U;
 
-        /// Train the model using data and outputs.
-        fn train(&mut self, data: &T, value: &U);
+        /// Train the model using inputs and targets.
+        fn train(&mut self, inputs: &T, outputs: &U);
```
Comment (AtheMathmo) [inline rusty-machine/src/lib.rs]: `outputs` in this function signature should be `targets`

File: rusty-machine/src/learning/nnet.rs [hunk 4]
```diff
@@ -211,13 +211,13 @@ impl<'a, T: Criterion> NeuralNet<'a, T> {
     */
 
     /// Compute the gradient using the back propagation algorithm.
-    fn compute_grad(&self, weights: &[f64], data: &Matrix<f64>, outputs: &Matrix<f64>) -> (f64, Vec<f64>) {
-        assert_eq!(data.cols(), self.layer_sizes[0]);
+    fn compute_grad(&self, weights: &[f64], inputs: &Matrix<f64>, outputs: &Matrix<f64>) -> (f64, Vec<f64>) {
```
Comment (AtheMathmo) [inline rusty-machine/src/learning/nnet.rs]: `outputs` in this signature should be `targets` to match the compute_grad function below.

File: rusty-machine/src/lib.rs [hunk 5]
```diff
@@ -145,8 +145,8 @@ pub mod learning {
 
             /// Return the optimized parameter using gradient optimization.
             ///
-            /// Takes in a set of starting parameters and related model data.
-            fn optimize(&self, model: &M, start: &[f64], data: &M::Data, outputs: &M::Target) -> Vec<f64>;
+            /// Takes in a set of starting parameters and related model Inputs.
+            fn optimize(&self, model: &M, start: &[f64], inputs: &M::Inputs, outputs: &M::Targets) -> Vec<f64>;
```
Comment (AtheMathmo) [inline rusty-machine/src/lib.rs]: `outputs` here should be called `targets` (to match the associated type).

I think this change needs to go through to the optimization algorithms themselves too.

File: rusty-machine/tests/learning/k_means.rs [hunk 6]
```diff
@@ -61,13 +61,13 @@ fn test_model_ran_partition() {
 #[test]
 fn test_model_kplusplus() {
     let mut model = KMeansClassifier::new(3);
-    let data = Matrix::new(3, 2, vec![1.0, 2.0, 1.0, 3.0, 1.0, 4.0]);
+    let inputs = Matrix::new(3, 2, vec![1.0, 2.0, 1.0, 3.0, 1.0, 4.0]);
     let pred_data = Matrix::new(3,2, vec![1.0, 2.0, 1.0, 3.0, 1.0, 4.0]);
```
Comment (AtheMathmo) [inline rusty-machine/tests/learning/k_means.rs]: Strictly speaking we should update `pred_data` to `targets` too. There's quite a few different terms present in the tests - that's my bad!

**Comment by AtheMathmo** [comment]
in the `k_means.rs` file you need to update the signature of the UnSupModel implementation. This means changing `data` to `inputs` in both cases.

**Comment by AtheMathmo** [comment]
The only remaining thing is the various naming issues in the tests (and perhaps in the documentation examples).

I'm happy to go through these myself and clean them up. I'm also happy to make the changes I detailed above if you'd prefer.

Thanks for your help!

**Comment by raulsi** [comment]
@AtheMathmo thanks for the review. and i would like to update this by myself. thanks though. :)

**Comment by raulsi** [comment]
Hey i just updated . please check and tell me if i miss something.

File: rusty-machine/src/learning/gp.rs [hunk 7]
```diff
@@ -184,24 +184,24 @@ impl<T: Kernel, U: MeanFunc> GaussianProcess<T, U> {
...
-            let mean = self.mean.func(data.clone());
+            let mean = self.mean.func(inputs.clone());
 
-            let post_mean = mean + self.ker_mat(data, t_data) * alpha;
+            let post_mean = mean + self.ker_mat(inputs, t_data) * alpha;
 
-            let test_mat = self.ker_mat(data, t_data);
-            let mut v_data = Vec::with_capacity(data.rows() * data.cols());
+            let test_mat = self.ker_mat(inputs, t_data);
+            let mut v_inputs = Vec::with_capacity(inputs.rows() * inputs.cols());
```
Comment (AtheMathmo) [inline rusty-machine/src/learning/gp.rs]: This shouldn't be `v_inputs`. By `v_data` I meant `variance_data`, i.e. the underlying data for the variance matrix.

**Comment by AtheMathmo** [comment]
There were some minor issues but I merged the PR and cleaned these up with my own commit. Thanks for all your help!

**Comment by raulsi** [comment]
Thanks. I would like to contribute more.  And is there anywhere we can chat and discuss

**Pull request merged**